internal/mining/mining_test.go

```diff
@@ -59,13 +59,62 @@ func TestNewBlockTemplateBasicErrorScenarios(t *testing.T) {
 	harness.chain.isTreasuryAgendaActiveErr = nil
 }
 
+// TestTipGenerationError tests the TipGeneration error scenario
+func TestTipGenerationError(t *testing.T) {
+	t.Parallel()
+
+	chainParams := chaincfg.MainNetParams()
+	chain := NewFakeChain(chainParams, chainParams.StakeValidationHeight-1)
+
+	tipGenerationErr := errors.New("error retrieving tip generation")
+	chain.tipGenerationErr = tipGenerationErr
+	harness, _, err := newMiningHarness(chainParams, chain)
+	if err != nil {
+		t.Fatalf("error creating mining harness: %v", err)
+	}
+
+	address, err := stdaddr.DecodeAddress("Dsi8CRt85xYyempXs7ZPL1rBxvDdAGZmgsg",
+		harness.chainParams)
+	if err != nil {
+		t.Fatalf("error decoding address: %v", err)
+	}
+
+	// Test that the error is triggered
+	_, err = harness.generator.NewBlockTemplate(address)
+	if err == nil {
+		t.Fatalf("expected error retrieving tip generation -- got nil")
+	} else if errorValue, ok := err.(Error); !ok {
+		t.Fatalf("expected error type `Error` -- got %v", err)
+	} else if errorValue.Err != ErrFailedToGetGeneration {
+		t.Fatalf("expected Err to be ErrFailedToGetGeneration -- got %v", errorValue.Err)
+	} else if errorValue.Description != "error retrieving tip generation" {
+		t.Fatalf("expected Description to be '%v' -- got '%v'", "error retrieving tip generation",
+			errorValue.Description)
+	}
+
+	// All else equal, if heightOfBestState is chainParams.StakeValidationHeight-2
+	// instead of chainParams.StakeValidationHeight-1, this error should not be
+	// triggered
+	chain = NewFakeChain(chainParams, chainParams.StakeValidationHeight-2)
+	chain.tipGenerationErr = tipGenerationErr
+	harness, _, err = newMiningHarness(chainParams, chain)
```

Comment:
I don't think there is a need to recreate the chain and harness.  Should be able to just update mocked state.

Reply:
See comment [here](https://github.com/santi-h/dcrd/pull/1#issuecomment-968505662) for why I was doing this. I can update.